email/add_ecr/email_sender.py

```diff
@@ -0,0 +1,311 @@
+        rating_change = cur.fetchall()
+        if not rating_change:
+            return 'No historical data for this author yet.'
+        return rating_change[1][0]
+    finally:
+        conn.close()
+
+
+def get_shelved_difference_from_yesterday(author_id) -> int:
+    '''gets shelved increase since yesterday for a given author_id'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+            SELECT 
+            shelved_count - LAG(shelved_count) OVER (
+                ORDER BY am.date_recorded DESC) shelved_increase_since_yesterday
+            FROM author AS a
+            JOIN author_measurement AS am ON a.author_id = am.author_id
+            WHERE am.author_id = {author_id}
+            ORDER BY am.date_recorded DESC
+            LIMIT 2
+            ;
+            '''
+        cur.execute(sql)
+        shelved_change = cur.fetchall()
+        if not shelved_change:
+            return 'No historical data for this author yet.'
+        return shelved_change[1][0]
+    finally:
+        conn.close()
+
+
+def get_todays_date_formatted() -> str:
+    '''Returns today's date formatted in y-m-d for use in the email's subject.'''
+    return datetime.today().strftime("%Y-%m-%d")
+
+
+def get_author_info(author_id: int) -> str:
+    '''Gets the author's information from their id'''
+    try:
+        conn = get_db_connection()
+        cur = conn.cursor()
+        sql = f'''
+        SELECT author_name, author_image_url
+        FROM author
+        WHERE author_id = {author_id}
+        '''
+        cur.execute(sql)
+        author_name, image_url = cur.fetchone()
+        return author_name, image_url
+    finally:
+        conn.close()
+
+
+def generate_author_html_container(author_id: int) -> str:
+    '''Generates a html table containing an author'''
+    author_name, image_url = get_author_info(author_id)
+    daily_shelved = get_shelved_difference_from_yesterday(author_id)
+    avg_rating_change = get_avg_rating_difference_since_yesterday(author_id)
+    container_html = f'''
+        <table width="100%" border="0" cellspacing="0" cellpadding="0"> 
+          <tr>
+            <td align="left" valign="top" width="150" style="padding: 10px;">
+              <img src="{image_url}" width="150" alt="A picture of the author_name" style="display: block; border-radius: 5px;">
+            </td>
+              <td align="left" valign="top" style="padding: 10px; font-family: Arial, sans-serif;">
+              <h3 style="margin: 0 0 10px 0; font-size: 18px; color: #2c3e50;">{author_name}'s engagement:</h3>
+              <p style="margin: 0; font-size: 14px; color: #555;"><p style="margin:5px 0 0 0;">
+              <strong>Daily Shelved:</strong> {daily_shelved} <br> 
+              <strong>Avg Rating Change:</strong> {avg_rating_change}
+              </p>
+            </td>
+          </tr>
+        </table>              
+        '''
+    return container_html
+
+
+def generate_html_head():
```

Comment:
Missing return type hint here
```suggestion
def generate_html_head() -> str:
```